lib/puppet/functions/to_python.rb

```diff
@@ -28,7 +28,7 @@ def to_python(object)
     case object
     when true then 'True'
     when false then 'False'
-    when :undef then 'None'
+    when nil then 'None'
```

Comment:
I had a talk with @hlindberg on slack when trying to understand which one I should use, and he advised to forget about `:undef` for new code.

Reply:
Adding to what I said in slack. The `:undef` is never produced by a modern puppet from the puppet language side when making function calls using the 4x function API. Before recent changes (in puppet 6 IIRC) transformation did take place of arguments to function calls to the deprecated 3x function API in accordance with how older versions of puppet worked (which was inconsistent using a mix of nil and :undef depending on if value was a top level argument value, or a nested value).

I wold expect `:undef` to only appear in logic when someone is calling a function from their custom ruby code and where they explicitly deal with `:undef` (pass it to the function they are calling).

The resource API and implementation has not undergone the same kind of scrubbing for `:undef` as the language/catalog-compilation side. Where handling of ´:undef` is still present in the language logic it is to compensate for the above (older functions, custom code not being up to date etc). On the "resource side" there are however no calls to puppet functions so that is not an issue.

The right thing to do is to not deal with `:undef` and instead use Ruby nil. There is however a small risk that the dropping of handling `:undef` in the input may affect someone.

Reply:
Thanks for the additional information. I always appreciate background.

> There is however a small risk that the dropping of handling `:undef` in the input may affect someone.

Given that this is a very new function, I'm not too concerned by anyone being affected.